File: client/src/app/modules/quotation/services/notice.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { NoticeType } from '../../quotation/model/NoticeType';

@Injectable({
  providedIn: 'root'
})
export class NoticeTypeService extends AppRestService<NoticeType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getNoticeTypes() {
    return this.getListCached(new HttpParams(), "notice-types");
  }

  addOrUpdateNoticeType(noticeType: NoticeType) {
    this.clearListCache(new HttpParams(), "notice-types");
    return this.addOrUpdate(new HttpParams(), "notice-type", noticeType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/NoticeTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.NoticeType;

public interface NoticeTypeService {
    public List<NoticeType> getNoticeTypes();

    public NoticeType getNoticeType(Integer id);

    public NoticeType addOrUpdateNoticeType(NoticeType noticeType);
}
